website/source/guides/secret-mgmt/versioned-kv.html.md

```diff
@@ -0,0 +1,950 @@
+```
+
+Once check-and-set is enabled, every write operation requires `cas` value to be
+passed. If you are sure that you want to overwrite the existing key-value, set
+`cas` to match the current version. Set `cas` to `0` if you want to write the
+secret _only if_ the key does not exists.
+
+**Example:**
+
+```shell
+# To write if the key does not already exists
+$ vault kv put -cas=0 secret/partner name="Example Co." partner_id="123456789"
+Key              Value
+---              -----
+created_time     2018-04-16T22:58:15.798753323Z
+deletion_time    n/a
+destroyed        false
+version          1
+
+# To overwrite the secret, you must specify the current version with -cas flag
+$ vault kv put -cas=1 secret/partner name="Example Co." partner_id="ABCDEFGHIJKLMN"
+Key              Value
+---              -----
+created_time     2018-04-16T23:00:28.66552289Z
+deletion_time    n/a
+destroyed        false
+version          2
+```
+
+#### API call using cURL
+
+To enable the **_Check-And-Set_** operation:
+
+```shell
+$ tee payload.json<<EOF
+{
+  "max_versions": 10,
+  "cas_required": true
+}
+EOF
+
+# Enable cas_requied on the secret engine mounted at secret/
+$ curl --header "X-Vault-Token: ..." \
+       --request POST \
+       --data @payload.json
+       http://127.0.0.1:8200/v1/secret/config
+
+# Enable cas_requied only on the secret/partner path
+$ curl --header "X-Vault-Token: ..." \
+      --request POST \
+      --data @payload.json
+      http://127.0.0.1:8200/v1/secret/metadata/partner
+```
+
+Once check-and-set is enabled, every write operation requires `cas` value to be
+passed. If you are sure that you want to overwrite the existing key-value, set
+`cas` to match the current version. Set `cas` to `0` if you want to write the
+secret _only if_ the key does not exists.
+
+**Example:**
+
+```shell
+# Write if the key does not already exists
+$ tee payload.json <<EOF
+{
+  "options": {
+    "cas": 0
+  },
+  "data": {
+    "name": "Example Co.",
+    "partner_id": "123456789"
+  }
+}
+EOF
+
+$ curl --header "X-Vault-Token: ..." \
+       --request POST \
+       --data @payload.json \
+       http://127.0.0.1:8200/v1/secret/data/partner
+
+# To overwrite the secret, you must pass the current version
+$ tee payload.json <<EOF
+{
+  "options": {
+    "cas": 1
+  },
+  "data": {
+    "name": "Example Co.",
+    "partner_id": "ABCDEFGHIJKLMN"
+  }
+}
+EOF
+
+$ curl --header "X-Vault-Token: ..." \
+       --request POST \
+       --data @payload.json \
+       http://127.0.0.1:8200/v1/secret/data/partner
+```
+
+<br>
+
+~> If the **`cas`** value is missing in your write request, the
+"`check-and-set parameter required for this call`" error will be returned.  If
+the `cas` does not match the current version number, you will receive the
+"`check-and-set parameter did not match the current version`" message.
+
+
+
+## Next steps
+
+This guide introduced the CLI commands and API endpoints to read and write
```

Comment:
This guide introduced the CLI commands and API endpoints to read and write secrets in the key-value secret engine.